### PR TITLE
Update `examples/api` for android platform

### LINKED ISSUE #147100: Examples in `examples/api` cannot run because of a missed step while migrating to the declarative plugins {} block
### Steps to reproduce

1. Open the `examples/api` project (master/beta/stable).
2. Run any example on the Android platform.

The issue is caused by this line in `examples/api/android/app/build.gradle`:
https://github.com/flutter/flutter/blob/2c473b33f335191e5f9f4b7028a0be46eff82d0c/examples/api/android/app/build.gradle#L67

It should be removed when migrating to the declarative plugins {} block.

This plugin should be moved to the bottom as well regarding to [this instruction](https://docs.flutter.dev/release/breaking-changes/flutter-gradle-plugin-apply).
https://github.com/flutter/flutter/blob/2c473b33f335191e5f9f4b7028a0be46eff82d0c/examples/api/android/app/build.gradle#L7

### Expected results

No issue found

### Actual results

```console
Warning: unable to detect project KGP version. Skipping version checking. 
This may be because you have applied KGP after the Flutter Gradle Plugin.

FAILURE: Build failed with an exception.

* Where:
Build file '/Users/thanhnhan/Documents/Development/Projects/Flutter/flutter/examples/api/android/app/build.gradle' line: 67

* What went wrong:
A problem occurred evaluating project ':app'.
> Could not get unknown property 'kotlin_version' for object of type org.gradle.api.internal.artifacts.dsl.dependencies.DefaultDependencyHandler.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s
Error: Gradle task assembleDebug failed with exit code 1
```

### Code sample

<details open><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[!] Flutter (Channel [user-branch], 3.22.0-14.0.pre.58, on macOS 14.3 23D56 darwin-arm64, locale en-VN)
    ! Flutter version 3.22.0-14.0.pre.58 on channel [user-branch] at /Users/thanhnhan/Documents/Development/Projects/Flutter/flutter
      Currently on an unknown channel. Run `flutter channel` to switch to an official channel.
      If that doesn't fix the issue, reinstall Flutter by following instructions at https://flutter.dev/docs/get-started/install.
    ! Upstream repository https://github.com/lamnhan066/flutter.git is not a standard remote.
      Set environment variable "FLUTTER_GIT_URL" to https://github.com/lamnhan066/flutter.git to dismiss this error.
    • Framework revision 663be6ff4a (29 minutes ago), 2024-04-20 09:26:47 +0700
    • Engine revision 3db86b3cab
    • Dart version 3.5.0 (build 3.5.0-81.0.dev)
    • DevTools version 2.35.0-dev.8
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update
      checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/thanhnhan/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/thanhnhan/Library/Android/sdk
    • Java binary at: /Users/thanhnhan/Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15E204a
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2023.1)
    • Android Studio at /Users/thanhnhan/Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)

[✓] IntelliJ IDEA Community Edition (version 2023.3.2)
    • IntelliJ at /Users/thanhnhan/Applications/IntelliJ IDEA Community Edition.app
    • Flutter plugin version 77.0.1
    • Dart plugin version 233.13135.65

[✓] VS Code (version 1.88.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.84.0

[✓] Connected device (5 available)
    • sdk gphone64 arm64 (mobile)     • emulator-5554             • android-arm64  • Android 14 (API 34) (emulator)
    • iPhone (mobile)                 • 00008101-000468C02608001E • ios            • iOS 17.4.1 21E236
    • macOS (desktop)                 • macos                     • darwin-arm64   • macOS 14.3 23D56 darwin-arm64
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad     • darwin         • macOS 14.3 23D56 darwin-arm64
    • Chrome (web)                    • chrome                    • web-javascript • Google Chrome 124.0.6367.61

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```

</details>

### PR DESCRIPTION
**Pull request by lamnhan066**
In the `examples/api`, the Android platform missed this step while migrating to the declarative plugins {} block described in [flutter-gradle-plugin-apply](https://docs.flutter.dev/release/breaking-changes/flutter-gradle-plugin-apply), so all examples of `examples/api` cannot run.

Fixes #147100

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

**Comment by Hixie** [comment]
Are these just regenerated from the `flutter create` template?

**Comment by lamnhan066** [comment]
@Hixie No, I removed the `kotlin_version` manually then I took a look at my new project and saw that we need to move the `dev.flutter.flutter-gradle-plugin` to the bottom.

**Review by bartekpacia — APPROVED** [review]
good catch, lgtm

**Review by reidbaker — CHANGES_REQUESTED** [review]
Hold on understanding what it takes to make your new test run in ci.

**Comment by lamnhan066** [comment]
@reidbaker  I found a line that related to the `examples/api` in [this log](https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8749988910130972353/+/u/run_test.dart_for_framework_tests_shard_and_subshard_misc/stdout) is

```
00:00 +0: loading /b/s/w/ir/x/w/flutter/examples/api/integration_test/smoke_integration_test.dart
```

Do you think the test didn't run because of the wrong name of the test file?

**Comment by reidbaker** [comment]
I just talked to some coworkers and the example API docs that use this code do not use the android or iOS sub directories which is why they were not tested. It turns out the correct fix is to delete the api/android and api/ios sub directories and add them to the local .gitignore. 

Do you want to update this or to do that or do you want to close this pr and have someone from the flutter team delete the code?

**Comment by lamnhan066** [comment]
@reidbaker  What should I do with the other platforms or do I just need to remove android and ios subdirectories?

**Comment by reidbaker** [comment]
> @reidbaker What should I do with the other platforms or do I just need to remove android and ios subdirectories?

You dont need to do anything other than delete android and ios and add those directories to gitignore to prevent them from being re-added accidentally. 

Quote from a teammate below. 
"our API docs we only use the dart code in examples/api/lib and run that code in an embedded dartpad. The platform directories are "unused" as far as I know. They are handy for local development of new samples (so you can just flutter run -d macos them). But for that just having support for desktop platforms is sufficient (hence the smoke test that only covers those platforms)."

**Comment by lamnhan066** [comment]
@reidbaker Done. Please give it a look.

File: .gitignore [hunk 1]
```diff
@@ -128,6 +128,10 @@ coverage/
 # Symbols
 app.*.symbols
 
+# Platform specific in the examples
```
Comment (reidbaker) [inline .gitignore:131]: Nit: "Unused platform specific files"
```suggestion
# Unused platform specific files
```

File: .gitignore [hunk 2]
```diff
@@ -128,6 +128,10 @@ coverage/
...
+examples/api/android/
```
Comment (reidbaker) [inline .gitignore:132]: Lets put these exclusions in the `/examples/api` directory as a new .gitignore file. The exclusions in this file are ones that should apply to anywhere in the codebase. 

When moved the contents would be 
```
# Unused platform specific files
android/
ios/
```

**Comment by lamnhan066** [comment]
Done

**Review by reidbaker — APPROVED** [review]
Thank you contributing to flutter!

**Comment by reidbaker** [comment]
I have added the autosubmit label. Once the tree is green a bot will merge this pr.

**Pull request merged**